dfastmi/gui/dialog_view.py

```diff
@@ -192,16 +192,22 @@ def _update_reach(self, data):
         self._update_condition_files()
 
     def _update_condition_files(self):
-        """Update the condition files."""
+        """
+        Update the condition files.
+        Use local copy of the dictionary because the callbacks from
+        the GUI updates will cause updates of the original dictionaries.
+        """
+        file_dictionary = self._view_model.reference_files.copy()
```

Comment:
Please do not please intelligence in the View. We have setup a MVVM principle for the GUI. Which means the data is stored somehow in the model and the view displays calculated data in the GUI from the ViewModel. Please move this code to the ViewModel. Display the results in the View. If you need help or an example I can assist.

Reply:
Discussed with Ralph, see PR #99

Reply:
Discussed with Bert.